### PR TITLE
Add Xcode 13.3.1 to test matrix.

### PR DESCRIPTION
**Pull request by ileitch**

### CODE REVIEW

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -15,8 +15,10 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        xcode: ["13.2.1", "13.1", "12.5.1", "12.4"]
+        xcode: ["13.3.1", "13.2.1", "13.1", "12.5.1", "12.4"]
         include:
+          - xcode: "13.3.1"
+            macos: macOS-11
```
Comment (liamnichols) [inline .github/workflows/test.yml:21]: Xcode 13.3 is only available on macOS 12 or later, but I can't tell if it's available yet or not.. https://github.com/github/roadmap/issues/240#issuecomment-1078329438 suggests that it is (also the readme is [here](https://github.com/actions/virtual-environments/blob/main/images/macos/macos-12-Readme.md)), but at the same time it is not listed on the main page yet: https://github.com/actions/virtual-environments. https://github.com/actions/virtual-environments/issues/3649 is also still open, so maybe you might need to request access to the private beta?

Comment (liamnichols) [reply .github/workflows/test.yml:21]: Ah yes, it's 'shipped' but only in private beta - https://github.com/github/feedback/discussions/13552

Looks like you can submit this form to request access: https://docs.google.com/forms/d/e/1FAIpQLScBYuIzySEU1AXVkK32gVoi95_dEz2qLNqt0QMnaJgoKMZ5YA/viewform?fbzx=-3839316793723587327

Comment (ileitch) [reply .github/workflows/test.yml:21]: Oh joy. I hoped GH would be on top of their upgrade game after the epicly slow rollout of macOS 11. Form submitted 🤞

**Review by liamnichols — APPROVED** [review]
macOS 12 is now available publicly so I re-ran the checks and they are passing 

https://github.com/actions/virtual-environments/pull/5438

I’ll merge this 👍

**Pull request merged**